### ISSUE TITLE
Exit early from "then" rule

### ISSUE THREAD

**Issue by chhatch** [elite-libs/rules-machine#41]
An object return rule will exit early with a value.
```js
const rules = [
  {
    return: '"blue"',
  },
  { return: '"green"' },
];

const result = ruleFactory(rules)();
console.log(result); // blue
```
But it will not work inside a "then" rule.
```js
const rules = [
  {
    if: 'true == true',
    then: {
      return: '"blue"',
    },
  },
  { return: '"green"' },
];

const result = ruleFactory(rules)();
console.log(result); // green
```
A string return rule can return a value but will not exit early.
```js
const rules = ['return "blue"', 'return "green"'];

const result = ruleFactory(rules)();
console.log(result); // green
```
Note: "return" is not currently an operator in our expression language. Try replacing "return" with "NickCage" if you don't believe me 😛 Unless.. Wait, @justsml did you alias "return" with "NickCage"?!

 In situations where we have many "if" rules but consider our operation done after any one of them is triggered, the ability to exit early with a literal value from a "then" rule would reduce cognitive overhead by eliminating an intermediate variable and simplifying our mental model. Consider this example adapted from @dara-rockwell 's recent issue.
```js
const things = [
  {
    name: 'ABC',
  },
  {
    name: 'DEF',
  },
  {
    name: 'GHI',
  },
  {
    name: 'JKL',
  },
];

const thingRules = [
  {
    if: 'STRING_CONTAINS("A", thing.name)',
    then: 'keep = false',
  },
  {
    if: 'STRING_CONTAINS("E", thing.name)',
    then: 'keep = false',
  },
  {
    if: 'STRING_CONTAINS("L", thing.name)',
    then: 'keep = false',
  },
  {
    return: 'keep',
  },
];

const filterWithRules = ruleFactory(thingRules);

function isFiltered(thing) {
  const result = filterWithRules({
    thing,
    keep: true,
  });
  return result;
}

console.dir(things.filter(isFiltered)); // [ { name: 'GHI' } ]
```
Now imagine that instead of 3 if statements it was 100. Even if we know that a particular element will trigger the first if statement we still have to look through the other 99 to know what result to expect. If we can exit early with a literal value the code is subtly changed but much easier to manage.
```js
const things = [
  {
    name: 'ABC',
  },
  {
    name: 'DEF',
  },
  {
    name: 'GHI',
  },
  {
    name: 'JKL',
  },
];

const thingRules = [
  {
    if: 'STRING_CONTAINS("A", thing.name)',
    then: 'return false',
  },
  {
    if: 'STRING_CONTAINS("E", thing.name)',
    then: 'return false',
  },
  {
    if: 'STRING_CONTAINS("L", thing.name)',
    then: 'return false',
  },
  {
    // or just 'return true'
    return: 'true',
  },
];

const filterWithRules = ruleFactory(thingRules);

function isFiltered(thing) {
  const result = filterWithRules({
    thing,
  });
  return result;
}

console.dir(things.filter(isFiltered)); // should be [ { name: 'GHI' } ]
```
